src/arrayMethodSort.js

```diff
@@ -4,8 +4,29 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (
+    compareFunction = function (a, b) {
+      return (
+        String(a).toLowerCase().localeCompare(String(b).toLowerCase()) ||
+        String(a).localeCompare(String(b)) ||
+        String(a)
+          .toLocaleUpperCase()
+          .localeCompare(String(b).toLocaleUpperCase())
+      );
+    },
+  ) {
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compareFunction(this[j], this[j + 1]) > 0) {
```

Comment:
When using a compare function, it should return a number less than 0, 0, or greater than 0. This implementation is correct, but ensure that the compareFunction follows this contract for all possible inputs.